### PR TITLE
Always create log directory if it doesn't exist

### PR DESCRIPTION
**Pull request by dschaper**
## Thank you for your contribution to the Pi-hole Community! 

Please read the comments below to help us consider your Pull Request.

We are all volunteers and completing the process outlined will help us review your commits quicker.

**Please make sure you**
 1. Base your code and PRs against the repositories developmental branch.
 2. [Sign Off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits as we enforce the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions

---
- **What does this PR aim to accomplish?:**

Symlinks can exist with a non-existent target. Be safe and always create the `/var/log/pihole/` directory if it doesn't exist.

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
- [x] I have read the above and my PR is ready for review. _Check this box to confirm_

### CODE REVIEW

**Review by PromoFaux — APPROVED** [review]

**Review by rdwebdesign — APPROVED** [review]

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -2796,6 +2796,11 @@ main() {
 
     stop_service pihole-FTL &> /dev/null
 
+    if [ -d /var/log/pihole/ ]; then
```
Comment (Joulinar) [inline automated install/basic-install.sh:2799]: `if [ ! -d /var/log/pihole/ ]; then`

Comment (MichaIng) [inline automated install/basic-install.sh:2799]:
```suggestion
    if [ ! -d /var/log/pihole/ ]; then
```
I'm sorry 😄.

Comment (dschaper) [reply automated install/basic-install.sh:2799]: No worries, thanks for the catch!

Comment (razorsedge) [reply automated install/basic-install.sh:2799]: But the PR didn't get the catch (at line 2799)?  Or am I missing something?

Comment (dschaper) [reply automated install/basic-install.sh:2799]: There's another PR that has the change.

**Pull request merged**